src/main/java/com/datatorrent/netlet/OptimizedEventLoop.java

```diff
@@ -59,7 +59,8 @@ public boolean add(SelectionKey key)
       if (key == null) {
         return false;
       }
-      if (pos >= keys.length) {
+      // An extra slot is needed to store the null delimiter when keys are retrieved
+      if (pos >= (keys.length - 1)) {
```

Comment:
Why wouldn't you just make both the fields public and use them instead of
having do nothing getter? That should speed everything up.

1. no null padding.
2. no checking for the padded null.
3. you can decrement the pos instead of incrementing it (supposedly faster
on x86 arch).
4. No unnecessary memory allocation just to accommodate the algo's
requirement.

--
Chetan



On Wed, Jul 29, 2015 at 8:56 AM, Vlad Rozov <notifications@github.com>
wrote:

> In src/main/java/com/datatorrent/netlet/OptimizedEventLoop.java
> <https://github.com/DataTorrent/Netlet/pull/11#discussion_r35776237>:
>
> > @@ -59,7 +59,8 @@ public boolean add(SelectionKey key)
> >        if (key == null) {
> >          return false;
> >        }
> > -      if (pos >= keys.length) {
> > +      // An extra slot is needed to store the null delimiter when keys are retrieved
> > +      if (pos >= (keys.length - 1)) {
>
> Good catch. To avoid subtraction can you move check and array copy after
> pos is incremented. What do you think of changing condition from >= to ==
> and using pos instead of key.length in arraycopy?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DataTorrent/Netlet/pull/11/files#r35776237>.
>